File: src/main/java/no/fintlabs/validation/MinTimestampBeforeMaxTimestamp.java
package no.fintlabs.validation;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = MinTimestampBeforeMaxTimestampValidator.class)
public @interface MinTimestampBeforeMaxTimestamp {

    String message() default "timestampMin must be before timestampMax";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/history/validation/MinTimestampBeforeMaxTimestamp.kt
package no.novari.flyt.history.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [MinTimestampBeforeMaxTimestampValidator::class])
annotation class MinTimestampBeforeMaxTimestamp(
    val message: String = "timestampMin must be before timestampMax",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
